src/calendar/tests/CalendarDate.spec.js

```diff
@@ -0,0 +1,460 @@
+import React from 'react/addons';
+import moment from 'moment';
+
+import CalendarDate from '../CalendarDate.jsx';
+
+import CalendarDatePeriod from '../CalendarDatePeriod.jsx';
+import CalendarHighlight from '../CalendarHighlight.jsx';
+import CalendarSelection from '../CalendarSelection.jsx';
+
+const TestUtils = React.addons.TestUtils;
+
+describe('The CalendarDate Component', function () {
+
+  const getCalendarDate = (props) => {
+
+    props = props || {};
+
+    return (<CalendarDate
+      date={props.date || moment()}
+      firstOfMonth={props.firstOfMonth || moment()}
+      dateRangesForDate={function () {
+        return {
+          count: function () {
+            return props.count || 1;
+          },
+          getIn: function (data) {
+            if (data[0] === 0) {
+              return '#333';
+            }
+            return '#444';
+          },
+        };
+      }}
+      onSelectDate={this.selectDateSpy}
+      onHighlightDate={this.highlightDateSpy}
+      onUnHighlightDate={this.unHighlightDateSpy}
+      isHighlightedDate={props.isHighlightedDate || false }
```

Comment:
Interesting comment. The implementation of the component indeed lacks getDefaultProps and would have made the setup of my tests easier :) Defaulting to testable values however would still be required in some cases I think, for instance here with dateRangesForDate.

I think the pattern you are suggesting here is much cleaner. I will refactor the code. I suggest leaving moving getDefaultProps into a different requirement as I wanted to avoid touching the source code.